hack/verify-typecheck.sh

```diff
@@ -28,32 +28,13 @@ kube::golang::verify_go_version
 
 cd "${KUBE_ROOT}"
 
-ret=0
-TYPECHECK_SERIAL="${TYPECHECK_SERIAL:-false}"
-
-SERVER_PLATFORMS=$(echo "${KUBE_SUPPORTED_SERVER_PLATFORMS[@]}" | tr ' ' ',')
-CLIENT_PLATFORMS=$(echo "${KUBE_SUPPORTED_CLIENT_PLATFORMS[@]}" | tr ' ' ',')
-NODE_PLATFORMS=$(echo "${KUBE_SUPPORTED_NODE_PLATFORMS[@]}" | tr ' ' ',')
-TEST_PLATFORMS=$(echo "${KUBE_SUPPORTED_TEST_PLATFORMS[@]}" | tr ' ' ',')
-
 # As of June, 2020 the typecheck tool is written in terms of go/packages, but
 # that library doesn't work well with multiple modules.  Until that is done,
 # force this tooling to run in a fake GOPATH.
+ret=0
+TYPECHECK_SERIAL="${TYPECHECK_SERIAL:-false}"
 hack/run-in-gopath.sh \
-    go run test/typecheck/main.go "$@" --serial="${TYPECHECK_SERIAL}" --platform "${SERVER_PLATFORMS}" "${KUBE_SERVER_TARGETS[@]}" \
-    || ret=$?
-hack/run-in-gopath.sh \
-    go run test/typecheck/main.go "$@" --serial="${TYPECHECK_SERIAL}" --platform "${CLIENT_PLATFORMS}" "${KUBE_CLIENT_TARGETS[@]}" \
-    || ret=$?
-hack/run-in-gopath.sh \
-    go run test/typecheck/main.go "$@" --serial="${TYPECHECK_SERIAL}" --platform "${NODE_PLATFORMS}" "${KUBE_NODE_TARGETS[@]}" \
-    || ret=$?
-
-# $KUBE_TEST_TARGETS doesn't seem to work like the other TARGETS variables...
-hack/run-in-gopath.sh \
-    go run test/typecheck/main.go "$@" --serial="${TYPECHECK_SERIAL}" --platform "${TEST_PLATFORMS}" test/e2e \
-    || ret=$?
-
+    go run test/typecheck/main.go "$@" "--serial=$TYPECHECK_SERIAL" || ret=$?
```

Comment:
re-expanding this check caught problems introduced in the last week, like non-OS-tagged files relying on interfaces only specified in OS-tagged files:

```
ERROR(windows/arm64): /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/conntrack/fake.go:34:7: undefined: Interface
ERROR(windows/amd64): /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/conntrack/fake.go:34:7: undefined: Interface
ERROR(windows/386): /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/conntrack/fake.go:34:7: undefined: Interface
```

Added a commit to fix that issue by moving the interface to a non-OS-specific file

Reply:
Thanks @liggitt !